pkg/boot/discovery.go

```diff
@@ -89,6 +89,8 @@ type Discovery interface {
 
 	// ListClusters lists the cluster names.
 	ListClusters(ctx context.Context) ([]string, error)
+	// GetClusterObject returns the dataSourceCluster object
+	GetClusterObject(ctx context.Context, cluster string) (*config.DataSourceCluster, error)
```

Comment:
Suggest rename the method, `GetClusterObject`  is confused with `GetCluster`.

`GetDataSourceCluster `  directly  or  other.